app/components/primer/alpha/navigation_list/item.rb

```diff
@@ -0,0 +1,121 @@
+
+module Primer
+  module Alpha
+    class NavigationList
+      # Items are rendered as styled links. They can optionally include leading and/or trailing visuals,
+      # such as icons, avatars, and counters. Items are selected based on their item_id, which is a
+      # required parameter. Items can also contain lists of subitems, which are rendered hidden but can
+      # be expanded on click.
+      class Item < Primer::Component
+        include ViewComponent::PolymorphicSlots
+
+        # The leading visual rendered before the link.
+        #
+        # @param kwargs [Hash] The arguments accepted by <%= link_to_component(Primer::Beta::Avatar) %> or <%= link_to_component(Primer::OcticonComponent) %>
+        renders_one :leading_visual, types: {
+          icon: Primer::OcticonComponent,
+          avatar: lambda { |**kwargs|
+            Primer::Beta::Avatar.new(**{ **kwargs, size: 16 })
+          }
+        }
+
+        # The trailing visual rendered after the link.
+        #
+        # @param kwargs [Hash] The arguments accepted by <%= link_to_component(Primer::OcticonComponent) %>, <%= link_to_component(Primer::LabelComponent) %>, or <%= link_to_component(Primer::CounterComponent) %>
+        renders_one :trailing_visual, types: {
+          icon: Primer::OcticonComponent,
+          label: Primer::LabelComponent,
+          counter: Primer::CounterComponent
+        }
+
+        # Sub-items are also links. They are indented slightly and rendered underneath the item.
+        #
+        # @param component_klass [Class] A custom component class to use instead of the default SubItem class.
+        # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+        renders_many :subitems, lambda { |component_klass: SubItem, **system_arguments|
+          component_klass.new(selected_item_id: @selected_item_id, **system_arguments)
+        }
+
+        # @param selected_by_ids [String, Array] The unique identifier or identifiers used to determine if the item is selected.
+        # @param href [String] The URL to link to.
+        # @param selected_item_id [String] The id of the currently selected item in the whole list. Can refer to an item or subitem.
+        # @param content_classes [String] Additional classes to add to the item's content, either an <a> or <span> tag.
+        # @param submenu_classes [String] Additional classes to add to each of the item's subitems, i.e. each <li> tag.
+        # @param disabled [Boolean] Whether or not the item is disabled. Disabled items are rendered as <span> tags instead of links.
+        # @param expanded [Boolean] Whether or not the list of subitems is expanded or collapsed.
+        # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+        def initialize(selected_by_ids: [], href: nil, selected_item_id: nil, content_classes: "", submenu_classes: "", disabled: false, expanded: false, **system_arguments)
+          @selected_by_ids = Array(selected_by_ids)
```

Comment:
why do we need `Array()`?

Reply:
It's not strictly necessary, but it is a nice convenience to allow folks to pass single arguments: `selected_by_ids: :an_id` or `selected_by_ids: [:an_id, :a_second_id]`. I'm open to removing that if you feel strongly about it. Let me know 😄